### PR TITLE
Created Memes routes and Controller

### PR DESCRIPTION
**Pull request by i-made-that**

### CODE REVIEW

File: client/index.html [hunk 1]
```diff
@@ -0,0 +1 @@
+<h1>HELLO WORLD!</h1>
```
Comment (ajkitson) [inline client/index.html]: Don't need this file

File: package.json [hunk 2]
```diff
@@ -14,8 +14,7 @@
     "test": "echo \"Error: no test specified\" && exit 1"
   },
   "repository": {
-    "type": "git",
-    "url": "https://github.com/fortunate-giraffe/memes-for-mankind.git"
+    "type": "git"
```
Comment (ajkitson) [inline package.json]: missed this one

File: server/config/middleware.js [hunk 3]
```diff
@@ -1,3 +1,21 @@
+var memeController = require('../memes/memeController.js');
+
 module.exports = function (app, express) {
+  // Express 4 allows us to use multiple routers with their own configurations
+  var memeRouter = express.Router();
```
Comment (ajkitson) [inline server/config/middleware.js]: what's up with memeRouter?

File: server/config/middleware.js [hunk 4]
```diff
@@ -1,3 +1,21 @@
...
+
   app.use(express.static(__dirname + '/../../client'));
+
+  app.get('/poo', function(req, res){
+    res.send('hello index');
```
Comment (ajkitson) [inline server/config/middleware.js]: creative

File: server/config/middleware.js [hunk 5]
```diff
@@ -1,3 +1,21 @@
...
+  });
+
+  app.get('/memes', memeController.getMemes);
+  app.get('/memes/create', memeController.createMeme);
+
+
+
+
+  // inject our routers into their respective route files
+  // require('../memes/memeRoutes.js')(memeRouter);
```
Comment (ajkitson) [inline server/config/middleware.js]: tidy up

File: server/memes/memeController.js [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    };
+    return http.get(options, function(res){
+      var body = '';
+      res.on('data', function(chunk){
+        body += chunk;
+      });
+      res.on('end', function(){
+        response.status(200).send(body);
+      });
+    });
```
Comment (ajkitson) [inline server/memes/memeController.js:22]: need an error handler, too

File: server/memes/memeController.js [hunk 7]
```diff
@@ -0,0 +1,72 @@
...
+      method: 'GET'
+    };
+    return http.get(options, function(res){
+      var body = '';
+      res.on('data', function(chunk){
+        body += chunk;
+      });
+      res.on('end', function(){
+        response.status(200).send(body);
+      });
```
Comment (ajkitson) [inline server/memes/memeController.js:21]: might want to check for error event

File: server/memes/memeController.js [hunk 8]
```diff
@@ -0,0 +1,72 @@
...
+    };
+
+    var req = http.request(options, function(res) {
+      console.log('STATUS: ' + res.statusCode);
+      console.log('HEADERS: ' + JSON.stringify(res.headers));
+      res.setEncoding('utf8');
+      res.on('data', function (chunk) {
+        var body = '';
+        body += chunk;
+        console.log('MY BODY', body);
```
Comment (ajkitson) [inline server/memes/memeController.js]: don't need both of these

File: server/memes/memeController.js [hunk 9]
```diff
@@ -0,0 +1,72 @@
...
+        response.status(200).send(body);
+      });
+    });
+  },
+
+  createMeme: function(request, response, next){
+
+    var postData = querystring.stringify({
+      'username': 'memesformankind',
+      'password': 'memesformankind',
```
Comment (ajkitson) [inline server/memes/memeController.js:29]: get these from a config file or env variables?

File: server/config/middleware.js [hunk 10]
```diff
@@ -1,3 +1,21 @@
...
+  var memeRouter = express.Router();
+
   app.use(express.static(__dirname + '/../../client'));
+
+  app.get('/poo', function(req, res){
+    res.send('hello index');
+  });
+
+  app.get('/memes', memeController.getMemes);
+  app.get('/memes/create', memeController.createMeme);
```
Comment (ajkitson) [inline server/config/middleware.js:12]: should this be a POST?

File: server/memes/memeController.js [hunk 11]
```diff
@@ -0,0 +1,72 @@
...
+      'generatorID': request.generator_id || 45,
+      'languageCode': 'en',
+      'imageID': request.imageID || 20,
+      'text0': request.text0 || 'toptext',
+      'text1': request.text1 || 'bottomtext'
```
Comment (ajkitson) [inline server/memes/memeController.js]: Will need to add bodyParser to get these properties. They'll then be on request.body.PROPERTY.

Also, the key names could be more intuitive

File: server/memes/memeController.js [hunk 12]
```diff
@@ -0,0 +1,72 @@
...
+        'Content-Length': postData.length
+      }
+    };
+
+    var req = http.request(options, function(res) {
+      console.log('STATUS: ' + res.statusCode);
+      console.log('HEADERS: ' + JSON.stringify(res.headers));
+      res.setEncoding('utf8');
+      res.on('data', function (chunk) {
+        var body = '';
```
Comment (ajkitson) [inline server/memes/memeController.js]: body should be outside the data event handler so it and the end event handler can access it via the closure. also need to add the end event handler and have it handle the response.send()

File: server/config/middleware.js (hunk 10 above)
Comment (i-made-that) [inline server/config/middleware.js:12]: Probably, but at the moment app.post() breaks the code.  Will have to check into this further!

**Pull request closed without merging**